waspc/src/Wasp/Generator/NpmDependencies.hs

```diff
@@ -141,9 +141,11 @@ combineNpmDepsForPackage npmDepsForWasp npmDepsForUser =
   if null conflictErrors && null devConflictErrors
     then
       Right $
+        -- todo(filip): check whether dependency updates and npm install work properly
+        -- todo(filip): reconsider whether we want to change the {sever,web-app}/package.json dynamically
         NpmDepsForPackage
-          { dependencies = waspDependencies npmDepsForWasp ++ remainingUserDeps,
-            devDependencies = waspDevDependencies npmDepsForWasp ++ remainingUserDevDeps
+          { dependencies = Map.elems remainingWapsDeps,
+            devDependencies = Map.elems remainingWaspDevDeps
```

Comment:
Let's discuss what we should do here. As a first step, of course. Option 1 is the better long-term solution.  
1. Always specify the same dependencies in `{sever,web-app}/package.json` and alias them. Wasp code will use our version of the dependencies, while the user's code will use their versions.
2. For each dependency, decide which version we allow.
   - If the user doesn't specify that dependency in their `package.json`, Wasp lists it in `{sever,web-app}/package.json`.
   - If the user specifies that dependency with the correct version in their `package.json`, Wasp installs it in the top-level `node_modules` and doesn't list it in `{sever,web-app}/package.json` (this is what this PR implements).
   - If the user specifies that dependency with the wrong version in their `package.json`, Wasp reports a conflict and refuses to build (this is what we do in the old structure).

Reply:
Hm I feel like I am missing information to make a decision here.

If option 1 is better long term, why not go with it now? I guess there are is a cost to implementation that solution vs what you implemented so far? What is that cost? What are pros/cons of each option?

In option 2, you said this PR doesn't list their dep in server|web-app/package.json since it will pull it from their node_modules instead. But you don't say what happens if they specify wrong version -> you said what happened before, not what happens now.

Reply:
Bump on this one!

Reply:
> If option 1 is better long term, why not go with it now? I guess there are is a cost to implementation that solution vs what you implemented so far?

Yes - I've never worked with package aliasing and would have to look into it. Also, we'd have to change more code than if we didn't go with it.

The implementation cost is the only con of this approach I currently know of, but there could be some caveats with aliased pacakges I'm not aware of.

> In option 2, you said this PR doesn't list their dep in server|web-app/package.json since it will pull it from their node_modules instead. But you don't say what happens if they specify wrong version -> you said what happened before, not what happens now.

I've edited my original comment to make the wording more clear. Sorry :upside_down_face:

Reply:
So you have option 2 working now, right? How much time do you estimate to get option 1 working? Option 1 sounds like it works better + is quite simpler, so it is clear winner, as you already said. I would say go for it, but obviously you would go for it already if you thought it is worth the time. So I am assuming instead you would like to do it a bit later? If so, I am also ok with that.
To conclude, I vote for option 1 but obviously you want to do it later so if so, sure.

Reply:
Ok, create an issue here: https://github.com/wasp-lang/wasp/issues/1644 .